### PR TITLE
Testing - packagingv2

### PR DESCRIPTION
**Pull request by Salamandar**
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by Salamandar** [comment]
@kay0u Could you please test this branch and tell me if everything is fine before merging ? <3

File: manifest.toml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    [resources.sources]
+    [resources.sources.go]
+    amd64.url = "https://go.dev/dl/go1.21.5.linux-amd64.tar.gz"
+    amd64.sha256 = "e2bc0b3e4b64111ec117295c088bde5f00eeed1567999ff77bc859d7df70078e"
+    i386.url = "https://go.dev/dl/go1.21.5.linux-386.tar.gz"
+    i386.sha256 = "8f4dba9cf5c61757bbd7e9ebdb93b6a30a1b03f4a636a1ba0cc2f27b907ab8e1"
+    arm64.url = "https://go.dev/dl/go1.21.5.linux-arm64.tar.gz"
+    arm64.sha256 = "841cced7ecda9b2014f139f5bab5ae31785f35399f236b8b3e75dff2a2978d96"
+    armhf.url = "https://go.dev/dl/go1.21.5.linux-armv6l.tar.gz"
+    armhf.sha256 = "837f4bf4e22fcdf920ffeaa4abf3d02d1314e03725431065f4d44c46a01b42fe"
```
Comment (kay0u) [inline manifest.toml:44]: Is there anything we can do to automagically upgrade these packages?

File: scripts/_common.sh [hunk 2]
```diff
@@ -96,6 +93,52 @@ ynh_remove_systemd_socket_config () {
...
 
+_ynh_add_dnsmasq() {
+    # Declare an array to define the options of this helper.
+    local legacy_args=t
+    local -A args_array=( [t]=template= )
+    local template
+    ynh_handle_getopts_args "$@"
+    local template="${template:-dnsmasq.conf}"
+
+    ynh_add_config --template="$template" --destination="/etc/dnsmasq.d/$app"
```
Comment (kay0u) [inline scripts/_common.sh:104]:
```suggestion
    ynh_add_config --template="dnsmasq.conf" --destination="/etc/dnsmasq.d/$app"
```

Let's remove the getops part

File: scripts/_common.sh [hunk 3]
```diff
@@ -96,6 +93,52 @@ ynh_remove_systemd_socket_config () {
...
+
+_ynh_add_ld_so() {
+    # Declare an array to define the options of this helper.
+    local legacy_args=t
+    local -A args_array=( [t]=template= )
+    local template
+    ynh_handle_getopts_args "$@"
+    local template="${template:-ld.so.conf}"
+
+    ynh_add_config --template="$template" --destination="/etc/ld.so.conf.d/$app.conf"
```
Comment (kay0u) [inline scripts/_common.sh:123]:
```suggestion
    ynh_add_config --template="ld.so.conf" --destination="/etc/ld.so.conf.d/$app.conf"
```

File: scripts/restore [hunk 4]
```diff
@@ -66,39 +29,31 @@ ynh_restore_file --origin_path="/usr/local/bin/lxd-migrate"
...
-
-echo "# Added by lxd
-root:100000:65536" | tee -a /etc/subuid /etc/subgid
-
 #=================================================
-# RESTORE SYSTEMD
+# RESTORE SYSTEM CONFIGURATIONS
 #=================================================
-ynh_script_progression --message="Restoring the systemd configuration..."
+ynh_script_progression --message="Restoring system configurations related to $app..." --weight=1
```
Comment (kay0u) [inline scripts/restore:35]:
```suggestion
ynh_script_progression --message="Restoring system configurations related to $app..."
```

File: scripts/upgrade [hunk 5]
```diff
@@ -47,45 +17,33 @@ fi
...
 
 ynh_systemd_action --service_name="$app.socket" --action="stop"
-ynh_systemd_action --service_name=$app --action="stop" --log_path="/var/log/$app/$app.log"
+ynh_systemd_action --service_name="$app" --action="stop" --log_path="/var/log/$app/$app.log"
 
 #=================================================
 # ENSURE DOWNWARD COMPATIBILITY
 #=================================================
-ynh_script_progression --message="Ensuring downward compatibility..."
+ynh_script_progression --message="Ensuring downward compatibility..." --weight=1
```
Comment (kay0u) [inline scripts/upgrade:25]:
```suggestion
ynh_script_progression --message="Ensuring downward compatibility..."
```

File: manifest.toml [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        "busybox",
+        "libattr1-dev",
+        "libuv1",
+        "libdevmapper-event1.02.1",
+        "dmeventd",
+        "lxc",
+        "tcl8.6",
+
+        # For debuging
+        "iproute2",
```
Comment (kay0u) [inline manifest.toml:96]:
```suggestion
        # According to https://documentation.ubuntu.com/lxd/en/latest/installing/#install-lxd-from-source
        "acl",
        "attr",
        "autoconf",
        "automake",
        "dnsmasq-base",
        "git",
        "libacl1-dev",
        "libcap-dev",
        "liblxc1",
        "liblxc-dev",
        "libsqlite3-dev",
        "libtool",
        "libudev-dev",
        "liblz4-dev",
        "libuv1-dev",
        "make",
        "pkg-config",
        "rsync",
        "squashfs-tools",
        "tar",
        "tcl",
        "xz-utils",
        "ebtables",

        # For lvm2
        "lvm2",
        "thin-provisioning-tools",

        # For btrfs
        "btrfs-progs"
```

**Comment by kay0u** [comment]
tested on my side, the main point now is about go, and how we can autoupdate it

**Comment by kay0u** [comment]
!testme

**Pull request merged**